pkg/kapp/app/recorded_app.go

```diff
@@ -297,9 +359,16 @@ func (a *RecordedApp) BeginChange(meta ChangeMeta) (Change, error) {
 }
 
 func (a *RecordedApp) update(doFunc func(*Meta)) error {
-	change, err := a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.name, metav1.GetOptions{})
+	change, err := a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.fqName, metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("Getting app: %s", err)
+		if errors.IsNotFound(err) {
+			change, err = a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.name, metav1.GetOptions{})
```

Comment:
Similar to `Exists()`, we might be able to use `meta()` to reduce one additional call.